modules/replicator/src/main/java/org/apache/ignite/internal/raft/client/TopologyAwareRaftGroupService.java

```diff
@@ -305,14 +305,16 @@ private static boolean recoverable(Throwable t) {
      * @return Future that is completed when all subscription messages to peers are sent.
      */
     public CompletableFuture<Void> subscribeLeader(LeaderElectionListener callback) {
-        assert !serverEventHandler.isSubscribed() : "The node already subscribed";
+        if (serverEventHandler.isSubscribed()) {
```

Comment:
That's is bit messy. The fact that we once set callback to serverEventHandler and if it was already initialised - add it to eventsClientListener while we also add serverEventHandler to eventsClientListener. That should be refactored. Please create jira and add TODO.

Reply:
Done